web/src/drafts.js

```diff
@@ -274,6 +275,62 @@ export function update_draft(opts = {}) {
 
 export const DRAFT_LIFETIME = 30;
 
+export function get_draft_ids_for_topic(stream_id, topic) {
+    const drafts = draft_model.get();
+    return Object.keys(drafts)
+        .filter(
+            (draft_id) =>
+                drafts[draft_id].type === "stream" &&
+                drafts[draft_id].stream_id === stream_id &&
+                drafts[draft_id].topic === topic,
+        )
+        .sort((draft_a, draft_b) => drafts[draft_a].updatedAt - drafts[draft_b].updatedAt);
+}
+
+export function get_draft_ids_for_private_message_recipient(private_message_recipient) {
+    const drafts = draft_model.get();
+    return Object.keys(drafts)
+        .filter(
+            (draft_id) =>
+                drafts[draft_id].type === "private" &&
+                _.isEqual(
+                    drafts[draft_id].private_message_recipient.split(",").sort(),
+                    private_message_recipient.split(",").sort(),
+                ),
+        )
+        .sort((draft_a, draft_b) => drafts[draft_a].updatedAt - drafts[draft_b].updatedAt);
+}
+
+export function restore_last_draft_based_on_compose_state() {
+    const message_type = compose_state.get_message_type();
+    if (!message_type || compose_state.message_content()) {
+        return;
+    }
```

Comment:
Let's make these two separate if statements, with comments noting when they are expected to be the case. 

Actually the `message_type` check might deserve to be an `assert`? `compose_state.set_message_type(msg_type);` is called just before this function.